WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginNoSitesFragment.kt

```diff
@@ -0,0 +1,152 @@
+    companion object {
+        const val TAG = "LoginNoSitesFragment"
+
+        fun newInstance(): LoginNoSitesFragment {
+            return LoginNoSitesFragment()
+        }
+    }
+
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    @Inject lateinit var meGravatarLoader: MeGravatarLoader
+    @Inject lateinit var uiHelpers: UiHelpers
+    private var loginListener: LoginListener? = null
+    private lateinit var viewModel: LoginNoSitesViewModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initDagger()
+        initBackPressHandler()
+        with(FragmentLoginNoSitesEmptyViewBinding.bind(view)) {
+            initContentViews()
+            initViewModel(savedInstanceState)
+        }
+    }
+
+    private fun initDagger() {
+        (requireActivity().application as WordPress).component().inject(this)
+    }
+
+    private fun FragmentLoginNoSitesEmptyViewBinding.initContentViews() {
+        buttonPrimary.setOnClickListener { viewModel.onSeeInstructionsPressed() }
+        buttonSecondary.setOnClickListener { viewModel.onTryAnotherAccountPressed() }
+    }
+
+    private fun FragmentLoginNoSitesEmptyViewBinding.initViewModel(savedInstanceState: Bundle?) {
+        viewModel = ViewModelProvider(this@LoginNoSitesFragment, viewModelFactory)
+                .get(LoginNoSitesViewModel::class.java)
+
+        initObservers()
+
+        viewModel.start(
+                application = requireActivity().application as WordPress,
+                savedInstanceState = savedInstanceState
+        )
+    }
+
+    private fun FragmentLoginNoSitesEmptyViewBinding.initObservers() {
+        viewModel.navigationEvents.observe(viewLifecycleOwner, { events ->
+            events.getContentIfNotHandled()?.let {
+                when (it) {
+                    is ShowSignInForResultJetpackOnly -> showSignInForResultJetpackOnly()
+                    is ShowInstructions -> showInstructions(it.url)
+                    else -> { // no op
+                    }
+                }
+            }
+        })
+
+        viewModel.uiModel.observe(viewLifecycleOwner, { uiModel ->
+            when (val state = uiModel.state) {
+                is ShowUser -> {
+                    loadGravatar(state.accountAvatarUrl)
+                    setUserName(state.userName)
+                    setDisplayName(state.displayName)
+                    loginEmptyViewUserSection.visibility = View.VISIBLE
+                }
+                is NoUser -> loginEmptyViewUserSection.visibility = View.GONE
+            }
+        })
+    }
+
+    private fun FragmentLoginNoSitesEmptyViewBinding.loadGravatar(avatarUrl: String) {
+        val url = meGravatarLoader.constructGravatarUrl(avatarUrl)
+        AvatarHelper.loadAvatarFromUrl(this@LoginNoSitesFragment, url, imageAvatar, object : AvatarRequestListener {
```

Comment:
Suggestion (💡): Since now we are at point were the 'loadAvatarFromUrl(...)` helper function doesn't need to `AvatarRequestListener`, I suggest making that optional and enhance this helper function with making the `AvatarRequestListener` nullable and default to `null`. This way this whole block can then be transformed to something more readable:
```
        AvatarHelper.loadAvatarFromUrl(
                fragment = this@LoginNoSitesFragment,
                avatarUrl = meGravatarLoader.constructGravatarUrl(avatarUrl),
                avatarView = imageAvatar
        )
```

Wdyt?

Reply:
I had thought about this, but didn't want to introduce null when others rely on it being non-nullable. Plus it's a Login library change and the risk was lower to use an empty handler. If you feel strongly, then I don't mind making the adjustment, just wanted you to be aware of why I didn't do it originally. @ParaskP7

Reply:
Yeap, yeap, I totally understand, actually I had the same feeling before commenting on that. Then, I realised that it is only logically that if the client of that api doesn't want to provide a listener to do something when the request is finished then, he should have to be forced by the api, exactly what is happening in this case. Thus, I though I should comment on enhancing the api. However, feel free to ignore, I am not feeling strongly about it. 😄 

PS: Just to get a bit more knowledge on it I would ask Ferrari on it and the reasoning behind the `listener` not being nullable. Probably, it was just due to it was being used everywhere it was exposed to.

Reply:
> Probably, it was just due to it was being used everywhere it was exposed to.

That would be my guess. Easy enough to ignore the listener if it's null in the `loadAvatarFromUrl` method, but how do we communicate back to the caller that it's required in those certain cases. Feels like an enhancement to the Login library and put the responsibility back on the user to decide HOW they want to use it.

Reply:
Yeap, it is indeed a bit of a 🤔 puzzle, if it were me I would be comfortable to put the responsibility back to the user, but let's ignore that for now to progress with it. It just might be worthy it to talk to Ferrari and add an enhancement issue for later on.